File: itsm-api/API/itsm/apps/cmdb/serializers/configurationitem.py
from rest_framework import serializers
from ..models.configurationItem import ConfigurationItem
from ..models.citype import CIType
from .ci_type import CITypeSerializer
from ...users.serializers.user import UserSerializer

class ConfigurationItemSerializer(serializers.ModelSerializer):
    ci_type = CITypeSerializer(read_only=True)
    owner = UserSerializer(read_only = True)

    class Meta:
        model = ConfigurationItem
        fields = [
            "id", 
            "name", 
            "ci_type", 
            "ci_type_id", 
            "status",  
            "serial_number",
            "owner",
            "owner_id"
        ]

        extra_kwargs = {
        'ci_type_id': {'source': 'ci_type', 'write_only': True},
        'owner_id': {'source': 'owner', 'write_only': True},
        }

File: itsm-front/my-app/src/interfaces/entities/ConfigurationItem.ts
import type ICIType from './CIType'
import type IUser from './User'

export default interface IConfigurationItem {
    id: number,
    name: string,
    status: "active" | "inactive" | "maintenance" | "retired",
    ci_type: ICIType,
    serial_number: string,
    owner: IUser;
}